test/integration/bench_memif_test.go

```diff
@@ -0,0 +1,173 @@
+package nsmd_integration_tests
+
+import (
+	"github.com/networkservicemesh/networkservicemesh/test/integration/nsmd_test_utils"
+	"github.com/networkservicemesh/networkservicemesh/test/kube_testing"
+	"github.com/networkservicemesh/networkservicemesh/test/kube_testing/pods"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	nscAgentName   = "vppagent-nsc"
+	icmpAgentName  = "vppagent-icmp-responder"
+	ipAddressParam = "IP_ADDRESS"
+	nscCount       = 5
+	nscMaxCount    = 15
+)
+
+func TestBenchMemifOneTimeConnecting(t *testing.T) {
+	RegisterTestingT(t)
+	if testing.Short() {
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+
+	Expect(err).To(BeNil())
+
+	k8s.PrepareDefault()
+
+	node := createNode(k8s)
+	vppAgentIcmp := k8s.CreatePod(pods.VppagentICMPResponderPod(icmpAgentName, node, icmpEnv()))
+	Expect(vppAgentIcmp.Name).To(Equal(icmpAgentName))
+
+	doneChannel := make(chan nscPingResult, nscCount)
+	defer close(doneChannel)
+
+	for count := nscMaxCount; count >= 0; count-- {
+		go createNscAndPingIcmp(k8s, count, node, vppAgentIcmp, doneChannel)
+	}
+
+	for count := nscMaxCount; count >= 0; count-- {
+		nscPingResult := <-doneChannel
+		Expect(nscPingResult.success).To(Equal(true))
+	}
+}
+
+func TestBenchMemifMovingConnection(t *testing.T) {
+	RegisterTestingT(t)
+	if testing.Short() {
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+
+	Expect(err).To(BeNil())
+
+	k8s.PrepareDefault()
+
+	node := createNode(k8s)
+
+	vppAgentIcmp := k8s.CreatePod(pods.VppagentICMPResponderPod(icmpAgentName, node, nscEnv()))
```

Comment:
Please check test_utils for a universal pod creation.

Reply:
fixed